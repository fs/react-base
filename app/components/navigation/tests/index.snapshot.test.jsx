import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { fakeSession, fakeEmptySession } from 'mocks/fakeSession';
import Navigation from 'components/navigation';

describe('Navigation', () => {
  let props;
  const renderComponent = () => renderer.create((
    <MemoryRouter>
      <Navigation {...props} />
    </MemoryRouter>
  ));

  beforeEach(() => {
    props = {
      ...fakeEmptySession,
      logout: () => {},
      signin: () => {},
      signup: () => {},
    };
  });

  it('renders correctly', () => {
    expect(renderComponent().toJSON()).toMatchSnapshot();
  });

  context('when user is logged in', () => {
    beforeEach(() => {
      props = {
        ...props,
        ...fakeSession,
      };
    });

    it('renders user navigations', () => {
      expect(renderComponent().toJSON()).toMatchSnapshot();
    });
  });
});
