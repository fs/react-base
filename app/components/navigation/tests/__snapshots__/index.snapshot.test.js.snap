// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
<nav
  className="panel navbar navbar-default"
>
  <div
    className="container"
  >
    <div
      className="navbar-header"
    >
      <span
        className="navbar-brand"
      >
        React-base
      </span>
    </div>
    <ul
      className="nav navbar-nav"
      role={null}
    >
      <li
        className="active"
        role="presentation"
        style={undefined}
      >
        <a
          aria-current="true"
          className="active"
          href="/"
          onClick={[Function]}
          onKeyDown={[Function]}
          style={Object {}}
        >
          Home
        </a>
      </li>
      <li
        className=""
        role="presentation"
        style={undefined}
      >
        <a
          aria-current={false}
          className={undefined}
          href="/about"
          onClick={[Function]}
          onKeyDown={[Function]}
          style={undefined}
        >
          About
        </a>
      </li>
    </ul>
    <ul
      className="nav navbar-nav navbar-right"
      role={null}
    >
      <li
        className=""
        role="presentation"
        style={undefined}
      >
        <a
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          Sign up
        </a>
      </li>
      <li
        className=""
        role="presentation"
        style={undefined}
      >
        <a
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          Sign in
        </a>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`Navigation when user is logged in renders user navigations 1`] = `
<nav
  className="panel navbar navbar-default"
>
  <div
    className="container"
  >
    <div
      className="navbar-header"
    >
      <span
        className="navbar-brand"
      >
        React-base
      </span>
    </div>
    <ul
      className="nav navbar-nav"
      role={null}
    >
      <li
        className="active"
        role="presentation"
        style={undefined}
      >
        <a
          aria-current="true"
          className="active"
          href="/"
          onClick={[Function]}
          onKeyDown={[Function]}
          style={Object {}}
        >
          Home
        </a>
      </li>
      <li
        className=""
        role="presentation"
        style={undefined}
      >
        <a
          aria-current={false}
          className={undefined}
          href="/about"
          onClick={[Function]}
          onKeyDown={[Function]}
          style={undefined}
        >
          About
        </a>
      </li>
    </ul>
    <ul
      className="nav navbar-nav navbar-right"
      role={null}
    >
      <li
        className=""
        role="presentation"
        style={undefined}
      >
        <a
          aria-current={false}
          className={undefined}
          href="/profile"
          onClick={[Function]}
          onKeyDown={[Function]}
          style={undefined}
        >
          test@test.com
        </a>
      </li>
      <li
        className=""
        role="presentation"
        style={undefined}
      >
        <a
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          Sign out
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
